lib/pluginmanager/proxy_support.rb

```diff
@@ -67,6 +68,9 @@ def configure_proxy
   proxy = get_proxy("http_proxy")
   if valid_proxy?(proxy)
     proxy_settings = extract_proxy_values_from_uri(proxy)
+    if proxy_settings[:host].nil?
```

Comment:
Since this PR introduces a bit more logic, I think the overall flow could be improved as there are some issues with it:
1. there's a valid_proxy? method that does little to validate the validity of the proxy :D
2. the `extract_proxy_values_from_uri` also does the uri parsing

Looking at the code:

```ruby
  proxy = get_proxy("http_proxy")
  if valid_proxy?(proxy)
    proxy_settings = extract_proxy_values_from_uri(proxy)
    if proxy_settings[:host].nil?
      raise StandardError.new("Invalid proxy `#{proxy}`. I parsed it to: #{proxy_settings}")
    end
    proxy_settings[:protocol] = "http"
    apply_env_proxy_settings(proxy_settings)
    proxies << proxy_settings
  end
```

We should probably have instead something like:
1. fetch string representing the proxy from env vars (`get_proxy`) - already done
2. check if proxy exists: probably rename `valid_proxy?` to `proxy_string_exists?` ?
3. parse the proxy string: instead of doing `URI.parse` in `extract_proxy_values_from_uri`, a new `parse_proxy_string` method could perform both the `URI.parse` and raise either a [ConfigurationError](https://github.com/elastic/logstash/blob/master/logstash-core/lib/logstash/errors.rb#L5) or [ArgumentError](https://apidock.com/ruby/ArgumentError) after performing a `.kind_of?(URI::HTTP)` check
4. pass the parsed URI object to `extract_proxy_values_from_uri` - already done

The kind_of check validates if the string is either an HTTP or HTTPS url:

```ruby
jruby-9.1.14.0 :016 > url = URI.parse("localhost:9399")
 => #<URI::Generic localhost:9399> 
jruby-9.1.14.0 :017 > url.kind_of?(URI::HTTP)
 => false 
jruby-9.1.14.0 :018 > url = URI.parse("https://localhost:9399")
 => #<URI::HTTPS https://localhost:9399> 
jruby-9.1.14.0 :019 > url.kind_of?(URI::HTTP)
 => true 
```

Reply:
Thanks for the review, I'll give this a try.